autosklearn/automl.py

```diff
@@ -917,6 +920,160 @@ def refit(self, X, y):
+        elif self._include_estimators is not None:
+            if task in CLASSIFICATION_TASKS:
+                include['classifier'] = self._include_estimators
+            elif task in REGRESSION_TASKS:
+                include['regressor'] = self._include_estimators
+            else:
+                raise ValueError(task)
+        elif self._exclude_estimators is not None:
+            if task in CLASSIFICATION_TASKS:
+                exclude['classifier'] = self._exclude_estimators
+            elif task in REGRESSION_TASKS:
+                exclude['regressor'] = self._exclude_estimators
+            else:
+                raise ValueError(task)
+
+        # Prepare missing components to the TAE function call
+        if 'include' not in kwargs:
+            kwargs['include'] = include
+        if 'exclude' not in kwargs:
+            kwargs['exclude'] = exclude
+        if 'memory_limit' not in kwargs:
+            kwargs['memory_limit'] = self._memory_limit
+        if 'resampling_strategy' not in kwargs:
+            kwargs['resampling_strategy'] = self._resampling_strategy
+        if 'metric' not in kwargs:
+            kwargs['metric'] = self._metric
+        if 'disable_file_output' not in kwargs:
+            kwargs['disable_file_output'] = self._disable_evaluator_output
+        if 'pynisher_context' not in kwargs:
+            kwargs['pynisher_context'] = self._multiprocessing_context
+        if 'stats' not in kwargs:
+            scenario_mock = unittest.mock.Mock()
+            scenario_mock.wallclock_limit = self._time_for_task
+            kwargs['stats'] = Stats(scenario_mock)
+        kwargs['stats'].start_timing()
+
+        # Allow to pass the cutoff also as an argument
+        cutoff = kwargs.pop('cutoff', self._per_run_time_limit)
+
+        # Fit a pipeline, which will be stored on disk
+        # which we can later load via the backend
+        ta = ExecuteTaFuncWithQueue(
+            backend=self._backend,
+            autosklearn_seed=self._seed,
+            abort_on_first_run_crash=False,
+            cost_for_crash=get_cost_of_crash(kwargs['metric']),
+            port=self._logger_port,
+            **kwargs,
+            **self._resampling_strategy_arguments
+        )
+
+        run_info, run_value = ta.run_wrapper(
+            RunInfo(
+                config=config,
+                instance=None,
+                instance_specific=None,
+                seed=self._seed,
+                cutoff=cutoff,
+                capped=False,
+            )
+        )
+
+        pipeline = None
+        if run_value.status == StatusType.SUCCESS:
+            if kwargs['resampling_strategy'] in ('cv', 'cv-iterative-fit'):
+                load_function = self._backend.load_cv_model_by_seed_and_id_and_budget
+            else:
+                load_function = self._backend.load_model_by_seed_and_id_and_budget
+            try:
+                pipeline = load_function(
```

Comment:
The pipeline doesn't have the data validator included, right? So we probably need to include that into the pipeline, too? Or we first aim to restructure Auto-sklearn to rely less on the data validator? Not sure here, maybe we can discuss this offline later.

Reply:
Please correct me if I miss something. I leverage on fit() for this. What I want to do is make this PR agnostic to the decision of what we do with the validator.

In this case, I call fit to build the datamanager (fit internally has the validator).